flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  description = "Solana development setup with Nix.";
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/c5e2528c7c4ec05ce05a563e3be64f3525b278ad";
```

Comment:
do we need to re-import these as flakes or can you simply use `pkgs.nix` here and only use flake-parts to create different systems? The reason is that it could cause a discrepancy between using flakes and non flake setup and the version are already fixed with niv sources so can simply import those inside this file (given there's no overriding of nixpkgs within additional sources, importing through flake inputs does not bring an additional advantage)

Reply:
sounds good, you can also import the pkgs.nix and pass it the system override for it to make the pkgs attrset

Reply:
Done with `nixfmt-tree`!

Reply:
I decided to unify the approach by:
* using the same revs as specified in the niv sources for the flake inputs
* using flake-compat to define pkgs.nix, thus having it consistent with the flake